src/rail/creation/degradation/observing_condition_degrader.py

```diff
@@ -324,16 +324,45 @@ def get_pixel_conditions(self, pixel: int) -> dict:
     def assign_pixels(self, catalog: pd.DataFrame) -> pd.DataFrame:
         """
         assign the pixels to the input catalog
+        check if catalogue contains position information;
+        if so, assign according to ra, dec;
+        else, assign randomly.
         """
         pixels = self.maps["pixels"]
         if "weight" in list((self.maps).keys()):
```

Comment:
FWIW, I think you can do 
`if "weight" in self.maps:`
here

Reply:
It seems like `weights` is only used in the `else` condition on line 346. So you could potentially move this if/else into a separate, `calculate_weights` function and only call `calculate_weights` after the `else` on line 346.

Reply:
Now moved the weights assignment after the `else` on line 346.